### PR TITLE
emoji-mart: update for v2.8.0

### PR DESCRIPTION
**Pull request by Nicell**
These changes update types for emoji-mart to support newer features.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
https://github.com/missive/emoji-mart/compare/v2.2.0...v2.8.0
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/emoji-mart/dist-es/components/category.d.ts [hunk 1]
```diff
@@ -1,12 +1,11 @@
-import React = require('react');
+import * as React from 'react';
```
Comment (Kovensky) [inline types/emoji-mart/dist-es/components/category.d.ts]: This was correct the other way around.

File: types/emoji-mart/dist-es/components/category.d.ts [hunk 2]
```diff
@@ -1,12 +1,11 @@
...
 
-import { EmojiData } from '..';
-
-import { Emoji, EmojiProps, I18n } from '.';
+import { Emoji, EmojiData, EmojiProps, I18n } from '..';
 
 export interface Props {
     emojis?: Array<string|EmojiData>;
     hasStickyPosition?: boolean;
+    id: string;
```
Comment (Kovensky) [inline types/emoji-mart/dist-es/components/category.d.ts]: This is [not in the `propTypes`](https://github.com/missive/emoji-mart/blob/v2.8.0/src/components/category.js#L213-L223) but that [looks like a bug](https://github.com/missive/emoji-mart/blob/v2.8.0/src/components/category.js#L147-L155) on the emoji-mart side.

I also questioned whether this was required or not but [looks like it is](https://github.com/missive/emoji-mart/blob/v2.8.0/src/components/category.js#L189)...

Comment (Kovensky) [reply types/emoji-mart/dist-es/components/category.d.ts]: Looks like it's supposed to be `keyof I18n['categories']`, or `import { CategoryName } from '../utils/shared-props'`. Probably better to add `CategoryName` to the reexports done in the entry point.

File: types/emoji-mart/dist-es/utils/shared-props.d.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    /** defaults to 52 */
+    sheetColumns?: number;
+    /** defaults to 52 */
+    sheetRows?: number;
+    /** defaults to 'apple' */
+    set?: EmojiSet;
+    size: number;
+    emoji: string | EmojiData;
+    html?: boolean;
+}
```
Comment (Kovensky) [inline types/emoji-mart/dist-es/utils/shared-props.d.ts]: [there's a `data` prop here that's listed as required](https://github.com/missive/emoji-mart/blob/v2.8.0/src/utils/shared-props.js#L4) but [it's in the default props](https://github.com/missive/emoji-mart/blob/v2.8.0/src/components/emoji/emoji.js#L19), which makes it an optional prop.

File: types/emoji-mart/emoji-mart-tests.tsx [hunk 4]
```diff
@@ -1,6 +1,6 @@
-// Port of https://github.com/missive/emoji-mart/blob/master/src/components/emoji.js
+// Port of https://missive.github.io/emoji-mart/
 
-import React = require('react');
+import * as React from 'react';
```
Comment (Kovensky) [inline types/emoji-mart/emoji-mart-tests.tsx]: Same, this was correct the other way around.

File: types/emoji-mart/emoji-mart-tests.tsx [hunk 5]
```diff
@@ -1,6 +1,6 @@
-// Port of https://github.com/missive/emoji-mart/blob/master/src/components/emoji.js
+// Port of https://missive.github.io/emoji-mart/
```
Comment (Kovensky) [inline types/emoji-mart/emoji-mart-tests.tsx]: This should link to the source code of the page; looks like it's been moved to https://github.com/missive/emoji-mart/blob/v2.8.0/stories/index.js

File: types/emoji-mart/dist-es/utils/shared-props.d.ts (hunk 3 above)
Comment (Kovensky) [reply types/emoji-mart/dist-es/utils/shared-props.d.ts]: However, the docs tell you to use `NimbleEmoji` instead if you want to pass custom data, so I guess it's best to omit it here; maybe with a comment.

File: types/emoji-mart/dist-es/components/picker/nimble-picker.d.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
```
Comment (Kovensky) [inline types/emoji-mart/dist-es/components/picker/nimble-picker.d.ts]: Keep all React imports as `import React = require('react')`. As long as react is [exported with `export =`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/aeb446e0bfc0ae94268414cd005ae52881b0a5ca/types/react/index.d.ts#L44) this is the only way of importing it that is correct regardless of the `esModuleInterop` setting.

**Review by Jessidhia — CHANGES_REQUESTED** [review]

File: types/emoji-mart/dist-es/utils/shared-props.d.ts (hunk 3 above)
Comment (Nicell) [reply types/emoji-mart/dist-es/utils/shared-props.d.ts]: Yeah I initially had data here, but then when testing had errors for obvious reasons. Will add comment.

File: types/emoji-mart/dist-es/components/picker/nimble-picker.d.ts (hunk 6 above)
Comment (Nicell) [reply types/emoji-mart/dist-es/components/picker/nimble-picker.d.ts]: Understood. Didn't know there was a reason behind this. Thanks for the info 😄

**Review by Jessidhia — APPROVED** [review]

**Pull request merged**